src/grizzly/services/webtransport/core.py

```diff
@@ -0,0 +1,126 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+import asyncio
+import os
+from asyncio import AbstractEventLoop
+from logging import getLogger
+from pathlib import Path
+from platform import system
+from threading import Thread
+from typing import Optional
+
+from aioquic.asyncio import serve  # type: ignore[attr-defined]
+from aioquic.h3.connection import H3_ALPN
+from aioquic.quic.configuration import QuicConfiguration
+
+from ..base import BaseService
+from .wpt_h3_server import webtransport_h3_server
+from .wpt_h3_server.webtransport_h3_server import (
+    SessionTicketStore,
+    WebTransportH3Protocol,
+    _connect_to_server,
+)
+
+# Override global _doc_root
+webtransport_h3_server._doc_root = os.path.join(  # pylint: disable=protected-access
+    os.path.dirname(os.path.abspath(__file__)),
+    "wpt_h3_server",
+    "handlers",
+)
+
+LOG = getLogger(__name__)
+
+
+class WebTransportServer(BaseService):
+    def __init__(self, port: int, cert: Path, key: Path) -> None:
+        """A WebTransport service.
+
+        Args:
+            port: The port on which to listen on.
+            cert: The path to the certificate file.
+            key: The path to the certificate's private key.
+        """
+        self._port = port
+        self._cert = cert
+        self._key = key
+
+        self._loop: Optional[AbstractEventLoop] = None
```

Comment:
Please use `| None` instead of `Optional`. This will require `from __future__ import annotations`

Reply:
Fixed in https://github.com/MozillaSecurity/grizzly/pull/484/commits/b26db8340f8f0e8463fed7dd42a57734e2b49f05